### ISSUE TITLE
Debugging task 'clean' always fails to start

### ISSUE THREAD

**Issue by dgard1981** [microsoft/vscode-azurefunctions#1328]
When attempting to debug an Azure Function App in Visual Studio, the 'clean' task fails -
> Error Message: Failed to start debugging. Task "clean" failed with exit code "1".

The terminal reports the following error. However, there is a csproj file in my working directory -
`MSBUILD : error MSB1003: Specify a project or solution file. The current working directory does not contain a project or solution file.
The terminal process terminated with exit code: 1`

Running 'dotnet clean' manually from the same working directory results in a successful build.

This issue seems to be [linked to another](https://github.com/Microsoft/vscode-azurefunctions/issues/810), but that was closed without a resolution due to lack of response, so I am not able to comment on it.

**Repro steps:**

I have been following the MS tutorial [here](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-function-vs-code).

1. In VS Code, open a folder that contains an Azure Function.
2. Press 'F5' to start debugging.

**Action:** azureFunctions.pickProcess
Error type: Error
Error Message: Failed to start debugging. Task "clean" failed with exit code "1".

**Version:** 0.17.1
**OS:** win32
**Product:** Visual Studio Code
**Product Version:** 1.34.0
**Language:** en

<details>
<summary>Call Stack</summary>

```
module.exports.t.pickFuncProcess.s.tasks.onDidEndTaskProcess.e extension.bundle.js:390:88446
u.fire extensionHostProcess.js:47:720
D.<anonymous> extensionHostProcess.js:629:772
s extensionHostProcess.js:13:221
```

</details>

**Comment by dgard1981** [comment]
OK, so I've found the issue...

After creating my Function App and a Function, I moved the files to a sub-directory (including _MyFunctionApp.csproj_ and the _MyFunction.cs_). Now to run the Function App, I `cd` to the sub-directory containing _MyFunctionApp.csproj_.

The problem now is that _.vscode/tasks.json_ is referencing the wrong directory, so the simple solution was to simply delete the _.vscode_ directory and then recreate it.

If that is not an option because of changes to _extensions.json_, _launch.json_ or _settings.json_, the following should get the debugger working. 

Add this to the **clean**, **build**, **clean release** and **publish** tasks -
```
"options": {
  "cwd": "${workspaceFolder}/My.Project"
}
```

And update the `cwd` in the **func** task to include the _My.Project_ directory.
Note: not all paths will be `/bin/Debug/newcoreapp2.1`, so don't just blindly copy.

**Old**

```
"options": {
  "cwd": "${workspaceFolder}/bin/Debug/netcoreapp2.1"
},
```

**New**

```
"options": {
  "cwd": "${workspaceFolder}/My.Project/bin/Debug/netcoreapp2.1"
},
```

While this isn't necessarily a bug, perhaps a better way to run tasks would be to target the PWD, rather than `${workspaceFolder}` - if that' possible. In a multi-project solution, this would then mean that you could run those tasks against whatever project you are currently working on.

**Comment by EricJizbaMSFT** [comment]
@dgard1981 I understand your frustration, although I'm not sure how best to handle the case where a user moves their project after the files in `.vscode` has been created.

> So while this isn't necessarily a bug, perhaps a better way to run tasks would be to target the PWD, rather than ${workspaceFolder} - if that' possible.

Here's the list of variables we can use: https://code.visualstudio.com/docs/editor/variables-reference I'm not sure if there's anything like the PWD you're talking about.

> The simple solution was to simply delete the .vscode directory and then have VS Code recreate it (next time you open that folder).

There's actually a simpler one (although I admit it's not very discoverable). You can run the command "Azure Functions: Inititialize for use with VS Code..." from the command palette and it should update just the relevant tasks/launch without breaking other tasks/launch (aka you don't have to delete the `.vscode` folder.)

In summary, glad you got it figured out. Let us know if you think of any improvements we could make, but otherwise we'll leave the experience as-is for now.
